fix hardcoded port in MultiBrowser LiveDevelopment and minor fix in PerfUtil

konradlipner opened the pull request:
If performance data gathering is disabled, markStart will return undefined instead of id. Calls like updateMeasurement and finalizeMeasurement will fail on id.id.

konradlipner committed:
PerfUtil fix errors when gathering is disabled

kolipka commented:
Travis says:
Running "cla-check-pull" task
Unexpected response from api.github.com
statusCode: 403
API rate limit exceeded for 52.0.240.122. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.

konradlipner committed:
fix hardcoded port in MultiBrowser LiveDevelopment

This cause LiveDev fail if port is already taken.
Now it can be changed via configuration.

abose commented:
We have a limit on the number of Travis builds that can go through before exceeding the Github API rate limit. It seems that dev activity was high during the time, and the rate limit exceeded. We usually retrigger the build after some time. So no worries in the case.

ingorichter commented on src/LiveDevelopment/MultiBrowserImpl/transports/node/NodeSocketTransportDomain.js:
```diff
@@ -58,7 +58,7 @@
     // This must match the port declared in NodeSocketTransport.js.
-    // TODO: randomize this?
+    // Default socket port number
     var SOCKET_PORT = 8123;
```
Since there is a pref for the socket port, this should be read from the prefs too.

kolipka replied:
Are you sure? This is part of NodeDomain, can I read prefs there? Anyway: starting server sends port number, this here will be overwritten.

konradlipner committed:
Merge remote-tracking branch 'upstream/master'

Conflicts:
	src/LiveDevelopment/MultiBrowserImpl/transports/NodeSocketTransport.js

konradlipner committed:
Merge remote-tracking branch 'upstream/master'

konradlipner committed:
Merge remote-tracking branch 'upstream/master'
